### DISCUSSION TITLE
Best practice at handling websocket backpresure at low level

### DISCUSSION THREAD

**Discussion by vlovch01** [uNetworking/uWebSockets#1427] [Q&A]
Hi,

I am using  uwebsocket for a server application and I started noticing that slow consuming clients 
are noticeable slowing down the server. 
Could someone point to best practice for handling back-pressure  so that no message is lost 
on client side?

Thank you.

**Comment by e3dio** [comment]
There was a guy who had problem with clients having too much backpressure, he reduced the message size by 10x by using binary encoding instead of JSON so there was 10x less data sent and 10x less backpressure and it seemed to solve the problem. Are your message sizes the smallest possible size? How much data sent per second?

**Reply by vlovch01** [reply]
I am already using binary messages. But some messages could be quite big. 
Good point is to make messages smaller.
